commit 7cd01b08d35f1b7d55686ed8cd57c94d3406ec8f
Author: Naveen N Rao <naveen@kernel.org>
Date:   Thu Jun 7 15:22:02 2018 +0530

    powerpc: Add support for function error injection
    
    We implement regs_set_return_value() and override_function_with_return()
    for this purpose.
    
    On powerpc, a return from a function (blr) just branches to the location
    contained in the link register. So, we can just update pt_regs rather
    than redirecting execution to a dummy function that returns.
    
    Signed-off-by: Naveen N. Rao <naveen.n.rao@linux.vnet.ibm.com>
    Reviewed-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>

diff --git a/arch/powerpc/lib/Makefile b/arch/powerpc/lib/Makefile
index 703afa1808ed..3bf9fc6fd36c 100644
--- a/arch/powerpc/lib/Makefile
+++ b/arch/powerpc/lib/Makefile
@@ -12,6 +12,8 @@ obj-y += string.o alloc.o code-patching.o feature-fixups.o
 
 obj-$(CONFIG_PPC32)	+= div64.o copy_32.o crtsavres.o strlen_32.o
 
+obj-$(CONFIG_FUNCTION_ERROR_INJECTION)	+= error-inject.o
+
 # See corresponding test in arch/powerpc/Makefile
 # 64-bit linker creates .sfpr on demand for final link (vmlinux),
 # so it is only needed for modules, and only for older linkers which